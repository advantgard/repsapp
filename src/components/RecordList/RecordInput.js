import React from "react";

export const RecordInput = props =>

	<input type="text" {...props} />;