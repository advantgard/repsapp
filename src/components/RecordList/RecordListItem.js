import React from "react";

export const RecordListItem = () => {

	return (
		<div className="record-table__item">
			ListItem
		</div>
	);

};