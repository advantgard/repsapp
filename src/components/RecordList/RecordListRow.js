import React from "react";

export const RecordListRow = (props) => {

	return (
		<li className="record-list__row">
			<div className="record-list__label-item"> {props.label} </div>
			<div className="record-list__percentage-item"> {props.percentage} </div>
		</li>
	);

};