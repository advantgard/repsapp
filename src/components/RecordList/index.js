export {RecordList} from "./RecordList";
export {RecordListRow} from "./RecordListRow";