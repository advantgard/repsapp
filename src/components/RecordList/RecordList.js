import React from "react";
import {RecordListRow} from "./RecordListRow";

export const RecordList = (props) => {

	return (
		<ul className="record-list">
			{props.items.map(item => <RecordListRow key={item.id} {...item} />)}
		</ul>
	);

};