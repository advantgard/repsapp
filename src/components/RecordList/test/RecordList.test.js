import React from "react";
import renderer from "react-test-renderer";
import { RecordList } from "../RecordList";

describe( "RecordList", () => {

	it( "renders correctly", () => {

		const tree = renderer.create(
			<RecordList record={160} steps={10}/>
		).toJSON();

		expect( tree ).toMatchSnapshot();

	} );

} );