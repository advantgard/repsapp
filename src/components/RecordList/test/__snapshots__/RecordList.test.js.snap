// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecordList renders correctly 1`] = `
<ul
  className="record-list"
>
  <li
    className="record-list__row"
  >
    <div
      className="record-list__label-item"
    >
       
      10%
       
    </div>
    <div
      className="record-list__percentage-item"
    >
       
      20 lbs
       
    </div>
  </li>
  <li
    className="record-list__row"
  >
    <div
      className="record-list__label-item"
    >
       
      20%
       
    </div>
    <div
      className="record-list__percentage-item"
    >
       
      35 lbs
       
    </div>
  </li>
  <li
    className="record-list__row"
  >
    <div
      className="record-list__label-item"
    >
       
      30%
       
    </div>
    <div
      className="record-list__percentage-item"
    >
       
      50 lbs
       
    </div>
  </li>
  <li
    className="record-list__row"
  >
    <div
      className="record-list__label-item"
    >
       
      40%
       
    </div>
    <div
      className="record-list__percentage-item"
    >
       
      65 lbs
       
    </div>
  </li>
  <li
    className="record-list__row"
  >
    <div
      className="record-list__label-item"
    >
       
      50%
       
    </div>
    <div
      className="record-list__percentage-item"
    >
       
      80 lbs
       
    </div>
  </li>
  <li
    className="record-list__row"
  >
    <div
      className="record-list__label-item"
    >
       
      60%
       
    </div>
    <div
      className="record-list__percentage-item"
    >
       
      100 lbs
       
    </div>
  </li>
  <li
    className="record-list__row"
  >
    <div
      className="record-list__label-item"
    >
       
      70%
       
    </div>
    <div
      className="record-list__percentage-item"
    >
       
      115 lbs
       
    </div>
  </li>
  <li
    className="record-list__row"
  >
    <div
      className="record-list__label-item"
    >
       
      80%
       
    </div>
    <div
      className="record-list__percentage-item"
    >
       
      130 lbs
       
    </div>
  </li>
  <li
    className="record-list__row"
  >
    <div
      className="record-list__label-item"
    >
       
      90%
       
    </div>
    <div
      className="record-list__percentage-item"
    >
       
      145 lbs
       
    </div>
  </li>
  <li
    className="record-list__row"
  >
    <div
      className="record-list__label-item"
    >
       
      100%
       
    </div>
    <div
      className="record-list__percentage-item"
    >
       
      160 lbs
       
    </div>
  </li>
</ul>
`;
